File: CS12B/pa3/DictionaryTest.java
//------------------------------------------------------------------------------
// Sudeep Baniya
// CMPS 12B
// sucbaniy
// 4-28-2016
// The programs that tests Dictionary.java
// DictionaryTest.java
//------------------------------------------------------------------------------
public class DictionaryTest{
      public static void main(String[] args){
      Dictionary test = new Dictionary();
      System.out.println(test.isEmpty());
      System.out.println(test);
      System.out.println(test.size());
      test.insert("2","f");
      System.out.println(test);
      test.insert("3","h");
      System.out.println(test);
      System.out.println(test.isEmpty());
      try{
      test.insert("2","f"); 
      }catch(DuplicateKeyException e){
      System.out.println("Caught Exception " + e);
      System.out.println("Continuing without interuption");
      }
      System.out.println(test);
      System.out.print(test.size());
      System.out.println();
      test.delete("2");
      System.out.println(test);
      System.out.print(test.size());
      System.out.println();
      test.makeEmpty();
      System.out.println(test);
      test.insert("1","z");
      System.out.println(test);
      test.insert("2","x");
      System.out.println(test);
      test.insert("3","3");
      System.out.println(test);
      test.insert("4","y");
      System.out.println(test);
      test.insert("5","m");
      System.out.println(test);
      test.insert("6","o");
      System.out.println(test);
      try{
      test.delete("50");
      }catch(KeyNotFoundException e){
      System.out.println("Caught Exception" + e);
      System.out.println("Continuing without interuption");
    }
  }
 } 

File: CS12B/lab5/DictionaryTest.c
//------------------------------------------------------------------------------
// Sudeep Baniya
// CMPS 12B
// sucbaniy
// 5-14-2016
// The programs test the Dictionary ADT based on the linked list data strucure.
// DictionaryTest.c
//------------------------------------------------------------------------------

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

#define MAX_LEN 180

int main(int argc, char* argv[]){
	Dictionary G = newDictionary();
	int b = size(G);
	int e = isEmpty(G);
	printf("%d,%d\n",b,e);
	char* k;
	char* v;
	char* word1[] = {"zero","one","two","three","four"};
	char* word2[] = {"foo", "bar", "blah", "galump"};
	int i;

	for (i=0; i<5; i++){
	insert(G, word1[i],word2[i]);
	}

	printDictionary(stdout, G);

	printf("%d,%d\n",b,e);

// insert (G, "zero", "lol"); //error: same key entered

	delete(G, "zero");
	delete(G, "one");
	delete(G, "two");
	insert(G, "five", "cmps");
	insert(G, "six", "data");

	printDictionary(stdout,G);
// delete(G, "zero"); // key not found error

	for ( i = 0; i<5; i++){
	k = word1[i];
	v = lookup(G,k);
	printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
	}

	printf("%s\n", (isEmpty(G)?"true":"false"));
	printf("%d\n", size(G));
	makeEmpty(G);
	printf("%s\n", (isEmpty(G)?"true":"false"));

	insert(G, "five", "cmps");
	insert(G, "six", "data");

	printDictionary(stdout,G);

	freeDictionary(&G);

	return(EXIT_SUCCESS);
	}
